### ISSUE TITLE
Failed to compile LSTM with --parallel

### ISSUE THREAD

**Issue by tungld** [onnx/onnx-mlir#2847]
I got an error when compiling the backend test lstm using `--parallel`:
```shell
$ ./Debug/bin/onnx-mlir --EmitLLVMIR -O3 -mcpu=z16 --parallel ../third_party/onnx/onnx/backend/test/data/node/test_lstm_defaults/model.onnx
loc("onnx.Transpose"("onnx.LSTM")): error: failed to legalize operation 'omp.loop_nest'
```

It looks related to parallelizing `onnx.Transpose` that was generated during the lowering of `onnx.LSTM` to krnl.

**Comment by tungld** [comment]
@AlexandreEichenberger any thought on this?

**Comment by AlexandreEichenberger** [comment]
@tungld let me investigate. 

It is pointing to the need to build with parallel enabled. I can add a lit test once I fix the bug.

**Comment by AlexandreEichenberger** [comment]
Interesting, the older mlir does this:
```
    omp.parallel {
      omp.wsloop  for  (%arg1) : index = (%c0) to (%c60) step (%c32) {
        %0 = vector.load %reshape[%arg1] : memref<60xf32>, vector<32xf32>
        %1 = math.absf %0 : vector<32xf32>
        vector.store %1, %reshape_2[%arg1] : memref<60xf32>, vector<32xf32>
        omp.yield
      }
      omp.terminator
    }
```

and the new mlir does this
```
    omp.parallel {
      omp.wsloop {
        omp.loop_nest (%arg1) : index = (%c0) to (%c60) step (%c32) {
          %0 = vector.load %reshape[%arg1] : memref<60xf32>, vector<32xf32>
          %1 = math.absf %0 : vector<32xf32>
          vector.store %1, %reshape_2[%arg1] : memref<60xf32>, vector<32xf32>
          omp.yield
        }
        omp.terminator
      }
      omp.terminator
    }
```

**Comment by AlexandreEichenberger** [comment]
Fixed by #2848

**Comment by tungld** [comment]
@AlexandreEichenberger Thank you very much!
